### ISSUE TITLE
Sort by RealmDictionary values

### ISSUE THREAD

**Issue by KunNiu** [realm/realm-java#7817]
### Problem

I created field RealmDictionary<Date>, and now I want to query all dates with a specified key, then sort by it's value. 
Now I query out all specified values, and then sort the list myself.
Can I complete the steps in one query?

### How important is this improvement for you?

I would like to have it but have a workaround

### Feature would mainly be used with

Local Database only

**Comment by KunNiu** [comment]
More extras, if I sort the list by self, then I cannot use the change listener, because index changed.

**Comment by rorbech** [comment]
Hi @KunNiu. We are actually working on supporting this in our query engine in https://github.com/realm/realm-core/pull/5311. Unfortunately it hasn't been finalized yet, but once done you should be able to achieve this with an raw predicate like:
```
realm.where(Object::class).rawPredicate("ANY dictionaryField.@keys == 'KEY' SORT(dictionaryField['key'] DESC)").findAll()
```

Until that, I don't really see a way to listen for notifications and correlate with a result sorted by Kotlin/Java in one operation. There are options to query only objects with a given field. You could maybe find a workaround where it is acceptable to just listen for the right updates and correlate with the sorted collection yourself by primary key or some other field.
```
val sortedObject = realm.where(Object::class).rawPredicate("ANY dictionaryField.@keys == 'KEY'").findAll().sortedBy {
   it.dictionaryField['KEY']
}
```

**Comment by rorbech** [comment]
@KunNiu We just released `10.16.0` that includes this new feature.
